src/transports/smart_protocol.c

```diff
@@ -108,6 +108,7 @@ static int append_symref(const char **out, git_vector *symrefs, const char *ptr)
 		if (giterr_last()->klass != GITERR_NOMEMORY)
 			goto on_invalid;
 
+		git__free(mapping);
```

Comment:
I think this should be part of the `on_invalid` label since that's where the code to free the contents of the refspec struct are.